readers/MoonReaderApp.py

```diff
@@ -0,0 +1,186 @@
+        self.annotations_db = "%s_imported_annotations" % self.app_name_
+        self.create_annotations_table(self.annotations_db)
+        self.books_db = "%s_imported_books" % self.app_name_
+        self.create_books_table(self.books_db)
+
+        self.annotated_book_list = []
+        self.selected_books = None
+
+        # Generate the book metadata from the selected book
+        row = self.opts.gui.library_view.currentIndex()
+        book_id = self.opts.gui.library_view.model().id(row)
+        db = self.opts.gui.current_db
+        mi = db.get_metadata(book_id, index_is_id=True)
+
+        tmpdir = tempfile.TemporaryDirectory()
+        with zipfile.ZipFile(raw, 'r') as mrpro:
+            mrpro.extractall(tmpdir.name)
+        db = self._db_location(tmpdir.name)
+        con = sqlite3.connect(os.path.join(tmpdir.name, app_package, db))
+        bookmap = {}
+        annotations = []
+        for row in con.execute("""
+                SELECT
+                    books._id,
+                    notes._id,
+                    notes.book,
+                    books.author,
+                    notes.time,
+                    notes.lastPosition,
+                    notes.highlightColor,
+                    notes.highlightLength,
+                    notes.original,
+                    notes.note
+                FROM
+                    notes
+                INNER JOIN
+                    books
+                        ON
+                            notes.filename = books.filename
+                ORDER BY
+                    time DESC;"""):
+
+            # Convert ms to seconds
+            timestamp = row[4] / 1000.0
+
+            if row[2] != mi.title and row[3] != mi.author:
+                continue
+
+            if row[0] not in bookmap:
+                book_mi = BookStruct()
+                book_mi.active = True
+                book_mi.book_id = mi.id
+                book_mi.title = row[2]
+                book_mi.author = row[3]
+                book_mi.uuid = None
+                book_mi.last_update = time.mktime(time.localtime())
+                book_mi.reader_app = self.app_name
+                book_mi.cid = mi.id
+                book_mi.annotations = 0
+                bookmap[row[0]] = book_mi
+
+            book = bookmap[row[0]]
+            book.annotations = book.annotations + 1
+
+            # Populate an AnnotationStruct
+            ann_mi = AnnotationStruct()
+            ann_mi.book_id = mi.id
+            ann_mi.last_modification = timestamp
+            ann_mi.timestamp = timestamp
+            ann_mi.location = row[5]
+            ann_mi.location_sort = row[5]
+            ann_mi.annotation_id = row[1]
+            # FIXME - colors in the moon reader db are integer values reflecting RGBA. I guess I'd need to figure out
+            #   an approximation of which color is being used. For now, everything's Yellow...
+            ann_mi.highlight_color = 'Yellow'
```

Comment:
Just to clarify: there's a limited set of colors I can use, right? I can't just put in an HTML hex color or something here?

Reply:
N/m, I got it working